WarshipGame/Assets/Scripts/Gameplay/Camera/CameraController.cs

```diff
@@ -0,0 +1,175 @@
+    private GameObject _focusGameObject;
+    private float _focusBoundaries;
+    private bool _isFocussing;
+    
+    [Space]
+    [Tooltip("The Cameras FOV will always be in between these 2 values.")]
+    [SerializeField] private Vector2 MaxZoom = new(35f,60f);
+    [Tooltip("The Speed in which the player can zoom the camera FOV.")]
+    [SerializeField] private float ZoomSpeed = 9;
+    [Tooltip("The standard zoomFOV for this camera controller.")] 
+    [SerializeField] private float DefaultFOVValue = 45;
+    private float _currentFOVValue;
+    
+    private void Awake()
+    {
+        MapSizeX *= 0.866f;
+        MapSizeZ = MapSizeZ * 0.866f - (0.866f / 2);
+        
+        _currentFOVValue = DefaultFOVValue;
+        CMVirtualCamera.m_Lens.FieldOfView = _currentFOVValue;
+    }
+
+    private void FixedUpdate()
+    {
+        PanCamera();
+
+        if (Input.GetAxis("Mouse ScrollWheel") != 0 && !_isFocussing)
+        {
+            ZoomCamera();
+        }
+    }
+
+    /// <summary>
+    /// This function makes it possible to drag the camera across the
+    /// play field between a certain boundary
+    /// </summary>
+    private void PanCamera()
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            _dragOrigin = Input.mousePosition;
+            _cameraOrigin = CMVirtualCamera.transform.position;
+        }
+
+        if (!Input.GetMouseButton(0)) return;
+        if (Input.GetAxis("Mouse X") == 0 && Input.GetAxis("Mouse Y") == 0) return;
+        
+        _dragDirection = Input.mousePosition;
+        Vector3 screenDelta = _dragOrigin - _dragDirection;
+
+        Vector2 screenSize = ScaleScreenToWorldSize(Cam.aspect, Cam.orthographicSize, Cam.scaledPixelWidth, Cam.scaledPixelHeight, screenDelta.x, screenDelta.y);
+
+        Vector3 move = new Vector3(screenSize.x + screenSize.y, 0f, screenSize.y - screenSize.x);
+        move = Quaternion.Euler(0,-90,0) * move;
+        Vector3 camPosMove = new Vector3(_cameraOrigin.x + move.x * Time.deltaTime * PanningSpeed, _cameraOrigin.y, _cameraOrigin.z + move.z * Time.deltaTime * PanningSpeed);
+
+        if (camPosMove.z <= -MapSizeZ || camPosMove.z >= MapSizeZ || camPosMove.x <= -MapSizeX || camPosMove.x >= MapSizeX)
+        {  
+            return;
+        }
+        
+        CMVirtualCamera.transform.position =  camPosMove;
+    }
+
+    /// <summary>
+    /// Corrects the screen panning by correcting the screen size and FOV distance
+    /// relative to the camera width and height
+    /// </summary>
+    /// <returns></returns>
+    private Vector2 ScaleScreenToWorldSize(float camAspect, float camFOV, float camScreenPixelWidth,
+        float camScreenPixelHeight, float screenW, float screenH)
+    {
+        float cameraWidth = camAspect * camFOV * 2f;
+        float cameraHeight = camFOV * 2f;
+        float screenWorldW = screenW * (cameraWidth / camScreenPixelWidth);
+        float screenWorldH = screenH * (cameraHeight / camScreenPixelHeight);
```

Comment:
Can you change W and H into Width and Height for easier understanding of the code